src/python_testing/TC_IDM_3_2.py

```diff
@@ -0,0 +1,182 @@
+from mobly import asserts
+from spec_parsing_support import build_xml_clusters
+
+class TC_IDM_3_2(MatterBaseTest, BasicCompositionTests):
+
+    @async_test_body
+    async def test_TC_IDM_3_2(self):
+
+        # Step 1
+
+        # TH selects a writeable attribute that is present on at least one endpoint.
+        # If no such attribute exists, skip this step. TH reads the attribute value and saves as `original`. TH sends a WriteRequestMessage to the DUT to write to the attribute on one endpoint.
+        #  It should set a value that is different than `original`.
+
+        # Verify that the DUT sends a WriteResponseMessage with any status except UNSUPPORTED_WRITE or DATA_VERSION_MISMATCH. If the Status is SUCCESS, verify the updated
+        #  value by sending a ReadRequestMessage for all affected paths. If the status is SUCCESS, send a WriteRequestMessage to set the value back to `original`.
+        self.user_params["use_pase_only"] = False
+        super().setup_class()
+        await self.setup_class_helper()
+        self.xml_clusters, self.problems = build_xml_clusters()
+        writable_attributes = []
+        for cluster_id in self.xml_clusters:
+            xml_cluster = self.xml_clusters[cluster_id]
+            attributes = xml_cluster.attributes
+            for attribute_id in attributes:
+                xml_attribute = attributes[attribute_id]
+                write_access = xml_attribute.write_access
+                if write_access:
+                    if cluster_id in Clusters.ClusterObjects.ALL_ATTRIBUTES:
+                        try:
+                            if hasattr(Clusters.ClusterObjects.ALL_ATTRIBUTES[cluster_id][attribute_id], 'value'):
+                                writable_attributes.append((cluster_id, attribute_id, write_access))
+                        except KeyError: # Some XML clusters aren't present in ALL_CLUSTERS or ALL_ATTRIBUTES
+                            pass
+        chosen_writable_attribute_tuple = writable_attributes[0]
+        chosen_writable_attribute = Clusters.ClusterObjects.ALL_ATTRIBUTES[chosen_writable_attribute_tuple[0]][chosen_writable_attribute_tuple[1]]
+        chosen_writable_cluster = Clusters.ClusterObjects.ALL_CLUSTERS[chosen_writable_attribute_tuple[0]]
+        
+        output_1 = await self.default_controller.ReadAttribute(self.dut_node_id, [chosen_writable_attribute])
+        endpoint = next(iter(output_1))
+        await self.default_controller.WriteAttribute(self.dut_node_id, [(endpoint, chosen_writable_attribute(value=123456))])
+        output_3 = await self.default_controller.ReadAttribute(self.dut_node_id, [chosen_writable_attribute])
+        
+        asserts.assert_not_equal(output_1[endpoint][chosen_writable_cluster][chosen_writable_attribute],
+                                 output_3[endpoint][chosen_writable_cluster][chosen_writable_attribute],
+                                 "Output did not change")
+

```

Comment:
need to add the cleanup step to return the attribute to its original value.

Reply:
Added a cleanup step by reading the initial value, saving it, and writing it back at the end of the step